pkg/vault/test/Authorizer.test.ts

```diff
@@ -27,7 +27,14 @@ describe('Authorizer', () => {
   const NOT_WHERE = ethers.Wallet.createRandom().address;
 
   sharedBeforeEach('deploy authorizer', async () => {
-    authorizer = await Authorizer.create({ admin });
+    const oldAuthorizer = await Authorizer.create({ admin });
+
+    vault = await deploy('Vault', { args: [oldAuthorizer.address, ZERO_ADDRESS, 0, 0] });
+    authorizer = await Authorizer.create({ admin, vault });
+
+    const setAuthorizerAction = await vault.getActionId(vault.interface.getSighash('setAuthorizer'));
+    await oldAuthorizer.grantPermissions(setAuthorizerAction, admin, vault, { from: admin });
+    await vault.connect(admin).setAuthorizer(authorizer.address);
```

Comment:
We will need to do all of this whenever we want to run a test that involves timelocks, correct? Otherwise we'd either have a circular dependency, or a zero vault in the authorizer which will cause all calls to `setDelay` to fail.

If that is true, then I wonder if we should try to either improve this or script this into a helper. Though perhaps that's unnecessary, since most (all?) timelock tests should live in this file.

Reply:
Yea, I don't think it would be useful for now... this is the only place where we do it